### PR TITLE
Add Photon packed parsing

### PR DESCRIPTION
**Pull request by mcm001**
Currently includes no checking of protocol version or anything

### CODE REVIEW

**Comment by jwbonner** [comment]
I restructured some things a bit, which should make it easier to add more raw schemas in the future. I also added support for WPILOGs, so users could now enable the default WPILib logging from NT on the RIO and get a log file with viewable Photon data. If you can confirm that the NT parsing still works as expected with the changes, we can get this merged.

File: src/hub/dataSources/schema/PhotonSchema.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+  return result;
+}
+
+/** Saves a pipeline result to a log file. */
+function saveResult(log: Log, key: string, timestamp: number, result: PhotonPipelineResult) {
+  log.putNumber(key + "/latency", timestamp, result.latency);
+  log.putNumber(key + "/timestamp", timestamp, result.timestamp);
+
+  for (const [idx, target] of result.targets.entries()) {
```
Comment (mcm001) [inline src/hub/dataSources/schema/PhotonSchema.ts:74]: LOL this curst code -- is there a more TS way to do this?

Comment (jwbonner) [reply src/hub/dataSources/schema/PhotonSchema.ts:74]: Well, you could do `result.targets.forEach(target => {})` and then explicitly have a log call for each field. For the target x/y a filter might be appropriate, like `value.detectedCorners.filter((element, index) => index % 2 == 0)`

**Comment by mcm001** [comment]
image
Works with datalogmanager logs too!

**Pull request merged**